tool/randomizer.lua

```diff
@@ -20,44 +20,89 @@ dependencies.beginner = {
+randomizer.queue = {{seed = ""}, {seed = "", difficulty = randomizer.difficulties[1]}}
+randomizer.newgamenewseed = "Auto"
+randomizer.newgamenewseedvalues = { "On", "Off", "Auto" }
+randomizer.newgamenewseedui = { ["On"] = "Always ON", ["Off"] = "Always OFF", ["Auto"] = "Auto (ON for Random Seed / OFF for Set Seed)" }
 
 local levelsequence
-local levelindex
+local levelindex = 1
+
+function randomizer.hudlines()
+    local currentseedline
+    local nextseedline = "Next: " .. randomizer.queue[2].difficulty .. " "
+    local progressline = ""
+
+    if randomizer.queue[1].seed == "" then -- If the user isn't currently playing a randomizer
+        currentseedline = "Press New Game to start"
+    else
+        currentseedline = "Current: " .. randomizer.queue[1].difficulty .. " " .. randomizer.queue[1].seedtype .. ": " .. randomizer.queue[1].seed
+        currentseedline = currentseedline .. "   Progress " .. levelindex - 2 .. "/" .. #levelsequence + 1
+    end
+
+    if randomizer.queue[2].seed == "" then -- If the next seed will be random
+        nextseedline = nextseedline .. randomizer.SEEDTYPE.RANDOMSEED
+    else
+        if randomizer.queue[1].seed == randomizer.queue[2].seed then -- If the next seed is the same as the current one
+            nextseedline = nextseedline .. "Keep current seed"
+        else
+            nextseedline = nextseedline .. randomizer.queue[2].seedtype .. ": " .. randomizer.queue[2].seed
+        end
+    end
 
-function randomizer.drawhud()
-    ui.drawlines({ "Randomizer " .. randomizer.proficiency .. " Seed " .. tostring(randomizer.seed) })
+    local randomizerlines = {currentseedline, nextseedline}
+    return randomizerlines
 end
 
 local function nextlevel()
     if levelindex <= #levelsequence then
         tas:set_level(levelsequence[levelindex])
-        levelindex = levelindex + 1
     end
+    levelindex = levelindex + 1
 end
 
 function randomizer.randomize()
-    if not randomizer.seed or randomizer.seed == "" then
-        randomizer.seed = os.time() .. math.floor(os.clock()*10000)
+    table.remove(randomizer.queue, 1)
+    if randomizer.queue[1].seed == "" then
+        randomizer.queue[1].seed = os.time() .. math.floor(os.clock()*10000)
+        if randomizer.newgamenewseed == "Off" then
+            table.insert(randomizer.queue, randomizer.queue[1])
+        else
+            table.insert(randomizer.queue, {seed = "", seedtype = randomizer.SEEDTYPE.RANDOMSEED, difficulty = randomizer.queue[1].difficulty})
+        end
+    else
+        if #randomizer.queue == 1 then
+            if randomizer.newgamenewseed == "On" then
+                table.insert(randomizer.queue, {seed = "", seedtype = randomizer.SEEDTYPE.RANDOMSEED, difficulty = randomizer.queue[1].difficulty})
+            else
+                table.insert(randomizer.queue, randomizer.queue[1])
+            end
+        end
```

Comment:
This if-else feels weird, complicated and complex to me. But I can't think of a better solution right now… I really want rust enums to represent
```rust
enum Run {
    SetSeed { seed: f64, difficulty: Difficulty },
    Random { difficulty: Difficulty },
}
```